pkg/binary/envoy/debug/node.go

```diff
@@ -42,76 +44,105 @@ func EnableNodeCollection(r *envoy.Runtime) {
 	r.RegisterPreTermination(activeConnections)
 }
 
+// Don't wait forever. This has hung on macOS before
+const processTimeout = 3 * time.Second
+
+// Errors from platform-specific libraries log to debug instead of raising an error or logging in an error category.
+// This avoids filling logs for unresolvable reasons.
 func ps(r binary.Runner) error {
 	f, err := os.Create(filepath.Join(r.DebugStore(), "node/ps.txt"))
 	if err != nil {
-		return fmt.Errorf("unable to create file to write ps output to: %v", err)
+		return fmt.Errorf("unable to create file to write ps output to: %w", err)
 	}
 	defer f.Close() //nolint
 
-	processes, err := process.Processes()
+	ctx, cancel := context.WithTimeout(context.Background(), processTimeout)
+	defer cancel()
+
+	processes, err := process.ProcessesWithContext(ctx)
+	if ctx.Err() == context.DeadlineExceeded {
+		log.Debugf("timeout getting a list of running processes")
+		return nil
+	}
 	if err != nil {
-		return fmt.Errorf("unable to get list of running processes: %v", err)
+		log.Debugf("unable to get list of running processes: %v", err)
+		return nil
 	}
-	return processPrinter(f, processes)
+
+	return printProcessTable(ctx, f, processes)
 }
 
-func processPrinter(out io.Writer, processes []*process.Process) error {
+func printProcessTable(ctx context.Context, out io.Writer, processes []*process.Process) error {
 	w := tabwriter.NewWriter(out, 0, 8, 5, ' ', 0)
 	fmt.Fprintln(w, "PID\tUSERNAME\tSTATUS\tRSS\tVSZ\tMINFLT\tMAJFLT\tPCPU\tPMEM\tARGS")
 	for _, p := range processes {
-		proc := safeProc(p)
-		fmt.Fprintf(w, "%v\t%v\t%v\t%v\t%v\t%v\t%v\t%.2f\t%.2f\t%v\n", proc.pid, proc.username, proc.status, proc.rss, proc.vms, proc.minflt,
+		proc := safeProc(ctx, p)
+		if proc == empty { // Ignore, but continue. The process could have died between the process listing and now.
+			continue
+		}
+		status := ""
+		if proc.status != nil && len(proc.status) > 0 {
```

Comment:
nit status is []string so it is safe to take len on nil:
```suggestion
		if len(proc.status) > 0 {
```